src/kernels/raw/launcher/kernelProcess.node.ts

```diff
@@ -277,8 +277,10 @@ export class KernelProcess implements IKernelProcess {
             this.exitEvent.fire({});
         });
         swallowExceptions(async () => {
-            if (this.connectionFile && (await this.fileSystem.localFileExists(this.connectionFile))) {
-                await this.fileSystem.deleteLocalFile(this.connectionFile);
+            if (this.connectionFile) {
+                await this.fileSystem
+                    .deleteLocalFile(this.connectionFile)
```

Comment:
Checking if the file exists, is even slower, then disposing, we don't care if the file exists or not, just delete and ignore the error (much faster, hence runs to completion when vscode is shutting down), else file never gets deleted